device/transport/mqtt/devdoc/mqtt_base_requirements.md

```diff
@@ -45,40 +46,61 @@ The `connect` method establishes a connection with the server using the config o
+**SRS_NODE_COMMON_MQTT_BASE_12_008: [** The `subscribe` method shall call `subscribe`  on MQTT.JS  library and pass it the `topic` and `options` arguments. **]**
+
+**SRS_NODE_COMMON_MQTT_BASE_16_023: [** The `subscribe` method shall throw a `ReferenceError` if the topic is falsy. **]**
+
+**SRS_NODE_COMMON_MQTT_BASE_16_024: [** The `subscribe` method shall call the callback with `null` and the `suback` object if the mqtt library successfully subscribes to the `topic`. **]**
+
+**SRS_NODE_COMMON_MQTT_BASE_16_025: [** The `subscribe` method shall call the callback with an `Error` if the mqtt library fails to subscribe to the `topic`. **]**
+
+**SRS_NODE_COMMON_MQTT_BASE_16_026: [** The `subscribe` method shall call the callback with a `NotConnectedError` if the connection hasn't been established prior to calling `publish`. **]**
 
-### Mqtt.publish(message)
-The `publish` method publishes the message passed as argument.
+### MqttBase.unsubscribe(topic, callback)
 
-**SRS_NODE_COMMON_MQTT_BASE_12_006: [** The `publish` method shall throw `ReferenceError` “Invalid message” if the message is falsy. **]**
+**SRS_NODE_COMMON_MQTT_BASE_16_031: [** The `unsubscribe` method shall throw a `ReferenceError` if the `topic` argument is falsy. **]**
 
-**SRS_NODE_COMMON_MQTT_BASE_12_007: [** The `publish` method shall call `publish`  on MQTT.JS  library with the given message. **]**
+**SRS_NODE_COMMON_MQTT_BASE_16_027: [** The `unsubscribe` method shall call the callback with a `NotConnectedError` if the connection hasn't been established prior to calling `publish`. **]**
 
-**SRS_NODE_COMMON_MQTT_BASE_16_008: [** The `publish` method shall use a topic formatted using the following convention: `devices/<deviceId>/messages/events/`. **]**
+**SRS_NODE_COMMON_MQTT_BASE_16_028: [** The `unsubscribe` method shall call `unsubscribe` on the mqtt library and pass it the `topic`. **]**
 
-**SRS_NODE_COMMON_MQTT_BASE_16_009: [** If the message has properties, the property keys and values shall be uri-encoded, then serialized and appended at the end of the topic with the following convention: `<key>=<value>&<key2>=<value2>&<key3>=<value3>(...)`. **]**
+**SRS_NODE_COMMON_MQTT_BASE_16_029: [** The `unsubscribe` method shall call the `callback` argument with no arguments if the operation succeeds. **]**
 
-**SRS_NODE_COMMON_MQTT_BASE_16_010: [** The `publish` method shall use QoS level of 1. **]**
+**SRS_NODE_COMMON_MQTT_BASE_16_030: [** The `unsubscribe` method shall call the `callback` argument with an `Error` if the operation fails. **]**
 
-**SRS_NODE_COMMON_MQTT_BASE_16_011: [** The `publish` method shall serialize the `messageId` property of the message as a key-value pair on the topic with the key `$.mid`. **]**
 
-**SRS_NODE_COMMON_MQTT_BASE_16_012: [** The `publish` method shall serialize the `correlationId` property of the message as a key-value pair on the topic with the key `$.cid`. **]**
+### MqttBase.updateSharedAccessSignature(sharedAccessSignature, callback)
```

Comment:
>e(sharedAccessSignature, callback) [](start = 39, length = 34)

I think there should be a requirement akin to the amqp: **SRS_NODE_DEVICE_AMQP_06_011: [** The `updateSharedAccessSignature` method shall call the `done` callback with a null error object and a SharedAccessSignatureUpdated object as a result, indicating the client does NOT need to reestablish the transport connection. **]**

Reply:
Yes, this will be created as part of the higher-level MQTT transport refactoring. 

---
In reply to: [140887707](https://github.com/Azure/azure-iot-sdk-node/pull/90#discussion_r140887707) [](ancestors = 140887707)